ignition-server/cmd/start.go

```diff
@@ -116,10 +116,25 @@ func setUpPayloadStoreReconciler(ctx context.Context, registryOverrides map[stri
 	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
 		return nil, fmt.Errorf("unable to set up ready check: %w", err)
 	}
-	if err = (&controllers.TokenSecretReconciler{
-		Client:       mgr.GetClient(),
-		PayloadStore: payloadStore,
-		IgnitionProvider: &controllers.LocalIgnitionProvider{
+
+	var ignProvider controllers.IgnitionProvider
+	switch cloudProvider {
+	case hyperv1.AWSPlatform:
```

Comment:
Or, we could make it so that the Local impl is only used on IBM cloud, and the old impl everywhere else?